diff --git a/Makefile b/Makefile
new file mode 100644
index 0000000..3e3afac
--- /dev/null
+++ b/Makefile
@@ -0,0 +1,8 @@
+export PYTHONPATH=../../lib/lib64/python2.6/site-packages/:../../lib/lib64/python
+
+install:
+       ./setup.py install --home ../../lib
+
+clean:
+       git reset --hard
+       git clean -f
diff --git a/pymongo/connection.py b/pymongo/connection.py
index 59710d8..f27afbd 100644
--- a/pymongo/connection.py
+++ b/pymongo/connection.py
@@ -40,6 +40,7 @@ import os
 import random
 import socket
 import struct
+import ssl
 import sys
 import threading
 import warnings
@@ -58,6 +59,7 @@ from pymongo.errors import (AutoReconnect,
 from pymongo.son import SON
 
 _CONNECT_TIMEOUT = 20.0
+_MONGODB_CERT_PATH = os.path.join(os.path.dirname(__file__), os.environ['ANYGIT_DB_CERT'])
 
 
 class Pool(threading.local):
@@ -438,6 +440,8 @@ class Connection(object): # TODO support auth for pooling
                     sock = socket.socket()
                     sock.setsockopt(socket.IPPROTO_TCP, socket.TCP_NODELAY, 1)
                     sock.settimeout(_CONNECT_TIMEOUT)
+                    sock = ssl.wrap_socket(sock, ca_certs=_MONGODB_CERT_PATH,
+                                           cert_reqs=ssl.CERT_REQUIRED)
                     sock.connect((host, port))
                     sock.settimeout(self.__network_timeout)
                     master = self.__master(sock)
